### DISCUSSION TITLE
Laravel Mix Asset Versioning

### DISCUSSION THREAD

**Discussion by ericp-mrel** [wintercms/winter#476] [Ideas]
I'm finding the new laravel mix integration kind of interesting and was wondering if there are any plans on adding support for [asset versioning](https://laravel-mix.com/docs/6.0/versioning). This would allow CSS/JS/images to bust the cache when they change since it's pretty common to have these types of files cached with a long cache lifetime.

It would be handy if there could be a `mix` twig filter added for this feature which would then look at the theme's `mix-manifest.json` file.
Example:
```twig
<link href="{{ 'assets/css/theme.css'|mix }}" rel="stylesheet"/>
```
I've actually created my own custom plugin which integrates Laravel mix versioning using twig filter, so it would be nice if this could be added to Winter CMS so that's one less plugin I have to maintain in all my sites.

**Comment by LukeTowers** [comment]
Sure, I'd welcome a PR to add mix() as a twig filter / function. Can be located in the System module. Target the wip/1.2 branch please 😉

**Reply by bennothommo** [reply]
@LukeTowers @ericp-mrel I'll add this shortly - I'm gonna need it for my own project. Gonna do a couple of QOL fixes for Snowboard at the same time.

**Reply by goldmont** [reply]
Hi there,

I would love it too. Any news about it? Thanks

**Reply by LukeTowers** [reply]
@jaxwilko ping

**Reply by jaxwilko** [reply]
@LukeTowers pong https://github.com/wintercms/winter/pull/1167

**Reply by LukeTowers** [reply]
@ericp-mrel @jaxwilko has done the hard bit of supporting mix manifest files in https://github.com/wintercms/winter/commit/7b181c24af70adcdc413ef0bd458e3faa2325798, feel free to submit a PR for a `| mix` filter.
